### PR TITLE
chore: update to React Query 4 stable

### PR DESCRIPTION
**Pull request by jxom**
## Description

Updating to React Query 4 stable!

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -105,9 +105,11 @@
   },
   "dependencies": {
     "@coinbase/wallet-sdk": "^3.3.0",
+    "@tanstack/query-sync-storage-persister": "^4.0.10",
+    "@tanstack/react-query": "^4.0.10",
+    "@tanstack/react-query-persist-client": "^4.0.10",
```
Comment (tmm) [inline packages/react/package.json:110]: Sad that these aren't entrypoints

**Review by tmm — APPROVED** [review]

File: packages/react/package.json (hunk 1 above)
Comment (jxom) [reply packages/react/package.json:110]: 💀

**Comment by holic** [comment]
image

I'm seeing a missing react-native peer dep error with pnpm after this, which feels...weird.

**Comment by jxom** [comment]
lol what

**Comment by jxom** [comment]
https://github.com/TanStack/query/blob/main/packages/react-query/package.json#L55
🤔

**Pull request merged**